tests/test_hsm_hotwallet.py

```diff
@@ -0,0 +1,267 @@
+    # Binance 8
+    return to_checksum_address("0xf977814e90da44bfa03b6295a0616a897441acec")
+
+
+@pytest.fixture()
+def anvil_eth_fork(request, weth_holder) -> str:
+    """Create a testable fork of Ethereum mainnet."""
+    mainnet_rpc = os.environ["ETH_NODE_URI"]
+    launch = fork_network_anvil(mainnet_rpc, unlocked_addresses=[weth_holder], fork_block_number=17_500_000)  # Pick a stable recent block
+    try:
+        yield launch.json_rpc_url
+    finally:
+        # Wind down Anvil process after the test is complete
+        launch.close(log_level=logging.ERROR)
+
+
+@pytest.fixture
+def web3(anvil_eth_fork: str):
+    """Set up a local unit testing blockchain."""
+    web3 = create_multi_provider_web3(anvil_eth_fork)
+    return web3
+
+
+@pytest.fixture()
+def deployer(web3) -> str:
+    """Deploy account.
+
+    Do some account allocation for tests.
+    """
+    return web3.eth.accounts[0]
+
+
+@pytest.fixture
+def hsm_wallet(web3: Web3) -> HSMWallet:
+    """HSM wallet implementation using env vars."""
+    wallet = HSMWallet()
+    wallet.sync_nonce(web3)
+    return wallet
+
+
+def test_eth_native_transfer(web3: Web3, deployer: str, hsm_wallet: HSMWallet):
+    """Test native ETH transfer using HSM wallet."""
+
+    # Fund HSM wallet with ETH
+    fund_tx = {"from": deployer, "to": hsm_wallet.address, "value": web3.to_wei(2, "ether")}
+    fund_tx_hash = web3.eth.send_transaction(fund_tx)
+    fund_receipt = web3.eth.wait_for_transaction_receipt(fund_tx_hash)
+    assert fund_receipt["status"] == 1
+
+    wallet_balance = web3.eth.get_balance(hsm_wallet.address)
+    logger.debug(f"HSM Wallet balance: {web3.from_wei(wallet_balance, 'ether')} ETH")
+    assert wallet_balance >= web3.to_wei(2, "ether")
+
+    # Prepare ETH transfer
+    recipient = "0x0000000000000000000000000000000000000000"
+    tx = {"from": hsm_wallet.address, "to": recipient, "value": web3.to_wei(1, "ether"), "gas": 21000, "gasPrice": web3.eth.gas_price, "chainId": web3.eth.chain_id}
+
+    # Sign and send
+    signed_tx = hsm_wallet.sign_transaction_with_new_nonce(tx)
+    tx_hash = web3.eth.send_raw_transaction(signed_tx.rawTransaction)
+    receipt = web3.eth.wait_for_transaction_receipt(tx_hash)
+    logger.debug(f"Transaction receipt: {receipt}")
+    assert receipt["status"] == 1
+
+    # Verify balances
+    final_recipient_balance = web3.eth.get_balance(recipient)
+    assert final_recipient_balance >= web3.to_wei(1, "ether")
+
+    final_wallet_balance = web3.eth.get_balance(hsm_wallet.address)
+    assert final_wallet_balance < web3.to_wei(1, "ether")  # Less than 1 ETH due to gas costs
+
+
+def test_dai_approval(web3: Web3, deployer: str, hsm_wallet: HSMWallet):
+    """Test DAI approve function with HSM wallet (DAI is simpler than USDC)."""
+
+    # Get DAI contract instead of USDC
+    # DAI on Ethereum mainnet
+    dai_details = fetch_erc20_details(web3, "0x6B175474E89094C44Da98b954EedeAC495271d0F")
```

Comment:
You do not need DAI on mainnet, but you can deploy your own token which is must faster. See here:

https://github.com/tradingstrategy-ai/web3-ethereum-defi/blob/9fdcb4b77bb6e886acfc769c105e08fb7bd979f6/tests/uniswap_v2/test_uniswap_v2_pair.py#L71